controller_manager:
  ros__parameters:
    update_rate: 1000  # Update rate in Hz

joint_state_broadcaster:
  type: joint_state_broadcaster/JointStateBroadcaster
  ros__parameters: {}

position_controller:
  type: position_controllers/JointTrajectoryController
  ros__parameters:
    joints:
      - robot_base
      - vertical_rail
      - prismatic_base
      - prismatic_arm
      - gripper_base
      - robotiq_85_right_knuckle_joint
      - robotiq_85_left_knuckle_joint
    state_publish_rate: 100
    command_publish_rate: 100
    command_interface: position  # Singular key
    state_interfaces:  # Ensure consistent indentation
      - position
      - velocity





